src/main/java/mezz/jei/api/IModRegistry.java

```diff
@@ -92,6 +92,19 @@
 	void addDescription(List<ItemStack> itemStacks, String... descriptionKeys);
 
 	/**
+	 * Adds an anvil recipe for the given inputs and output.
+	 * @param leftInput The itemStack placed on the left slot.
+	 * @param rightInput The itemStack placed on the right slot.
+	 * @param output The resulting itemStack.
+	 * @param levelsCost The cost, in experience levels, that will be required in order to pick up the item.
+	 *                   Optional: If not specified, the cost text will not display on the recipe.
+	 */
+	void addAnvilRecipe(ItemStack leftInput, ItemStack rightInput, ItemStack output, int levelsCost);
+	void addAnvilRecipe(ItemStack leftInput, ItemStack rightInput, ItemStack output);
+	void addAnvilRecipe(ItemStack leftInput, List<ItemStack> rightInput, List<ItemStack> output, int levelsCost);
+	void addAnvilRecipe(ItemStack leftInput, List<ItemStack> rightInput, List<ItemStack> output);

```

Comment:
I think that makes the use of the API unnecessarily ugly, but your choice.
However, for context, the option to not have a cost shown wasn't about the recipe not having a cost, it was about the recipe's cost not being known beforehand, such as for enchanted item repair, where the cost depends on how many times the item has been repaired, and how many enchants it has.